exchanges/account/account.go

```diff
@@ -292,6 +302,82 @@ func (s *Service) Save(incoming *Holdings, creds *Credentials) error {
 	return errs
 }
 
+// Update updates the balance for a specific exchange and credentials
+func (s *Service) Update(exch string, changes []Change, creds *Credentials) error {
+	if exch == "" {
+		return fmt.Errorf("cannot update balance: %w", errExchangeNameUnset)
+	}
+
+	if creds.IsEmpty() {
+		return fmt.Errorf("cannot update balance: %w", errCredentialsAreNil)
+	}
+
+	exch = strings.ToLower(exch)
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	accounts, exist := s.exchangeAccounts[exch]
+	if !exist {
+		id, err := s.mux.GetID()
+		if err != nil {
+			return err
+		}
+		accounts = &Accounts{
+			ID:          id,
+			subAccounts: make(map[Credentials]map[key.SubAccountAsset]currencyBalances),
+		}
+		s.exchangeAccounts[exch] = accounts
+	}
+
+	var subAccounts map[key.SubAccountAsset]currencyBalances
+	subAccounts, exist = accounts.subAccounts[*creds]
+	if !exist {
+		subAccounts = make(map[key.SubAccountAsset]currencyBalances)
+		accounts.subAccounts[*creds] = subAccounts
+	}
+
+	var errs error
+	for _, change := range changes {
+		if !change.AssetType.IsValid() {
+			errs = common.AppendError(errs, fmt.Errorf("cannot update balance for %s [%s] %w",
+				change.Account,
+				change.AssetType,
+				asset.ErrNotSupported))
+			continue
+		}
+
+		accAsset := key.SubAccountAsset{
+			SubAccount: change.Account,
+			Asset:      change.AssetType,
+		}
+		assets, exist := subAccounts[accAsset]
+		if !exist {
+			assets = make(map[*currency.Item]*ProtectedBalance)
+			accounts.subAccounts[*creds][accAsset] = assets
+		}
+		if change.Balance != nil {
+			bal, ok := assets[change.Balance.Currency.Item]
+			if !ok || bal == nil {
+				bal = &ProtectedBalance{}
+				assets[change.Balance.Currency.Item] = bal
+			}
+
+			if err := bal.load(change.Balance); err != nil {
+				errs = common.AppendError(errs, fmt.Errorf("cannot update balance for %s [%s %s] %w",
+					change.Account,
+					change.AssetType,
+					change.Balance.Currency,
+					err))
+				continue
+			}
+		}
```

Comment:
If a developer has not set a `Balance` pointer, that seems like an issue to note.
```go
if change.Balance == nil {
    errs = common.AppenedError(errs, "the thing for exchange %s and asset %s is bad")
    continue
}
```